backend/src/parameter_parser.py

```diff
@@ -107,9 +108,11 @@ def process_search_params(params: ParamContainer) -> SearchParams:
                      vaccine_type is not None]
     if len(vaccine_types) < 1:
         raise ParameterException(f"Search parameter {immunization_target_key} must have one or more values.")
-    if any([x not in VaccineTypes().all for x in vaccine_types]):
+
+    valid_vaccine_types = redis_client.hkeys(VACCINE_TYPE_TO_DISEASES_HASH_KEY)
+    if any([x not in valid_vaccine_types for x in vaccine_types]):
```

Comment:
Minor point but it resolves a Sonar warning:
```suggestion
    if any(x not in valid_vaccine_types for x in vaccine_types):
```